bluesky_queueserver/manager/tests/test_qserver_cli.py

```diff
@@ -1054,3 +1056,91 @@ def test_qserver_reload_permissions(re_manager_pc_copy, tmp_path):  # noqa F811
 
     # Attempt to add the plan to the queue. It should be successful now.
     assert subprocess.call(["qserver", "queue", "add", "plan", plan]) == SUCCESS
+
+
+# fmt: off
+@pytest.mark.parametrize("test_mode", ["none", "ev", "cli"])
+# fmt: on
+def test_qserver_secure_1(monkeypatch, re_manager_cmd, test_mode):  # noqa: F811
+    """
+    Test operation of `qserver` CLI tool with enabled encryption. Test options to
+    set the private key used by `qserver` using CLI parameter and environment variable.
+    """
+    public_key, private_key = generate_new_zmq_key_pair()
+
+    if test_mode == "none":
+        # No encryption
+        cli_p = []
+    elif test_mode == "ev":
+        # Set server public key (for 'qserver') using environment variable
+        cli_p = []
+        monkeypatch.setenv("QSERVER_ZMQ_PUBLIC_KEY", public_key)
+        # Set private key for RE manager
+        monkeypatch.setenv("QSERVER_ZMQ_PRIVATE_KEY", private_key)
+        # Set public key used by test helper functions such as 'wait_for_condition'
+        set_qserver_zmq_public_key(monkeypatch, server_public_key=public_key)
+    elif test_mode == "cli":
+        # Set server public key (for 'qserver') using CLI parmeter
+        cli_p = ["--zmq-public-key", public_key]
+        # Set private key for RE manager
+        monkeypatch.setenv("QSERVER_ZMQ_PRIVATE_KEY", private_key)
+        # Set public key used by test helper functions such as 'wait_for_condition'
+        set_qserver_zmq_public_key(monkeypatch, server_public_key=public_key)
+    else:
+        raise RuntimeError(f"Unrecognized test mode '{test_mode}'")
```

Comment:
This code is repeating in `test_zmq_api.py`, maybe it's worth extracting it and placing it somewhere to be reused?

Reply:
Not critical for now.